public/js/selfoss-db.js

```diff
@@ -136,23 +270,534 @@ selfoss.dbOnline = {
+                            .filter(function(entry) {
+                                return !entry.unread && !entry.starred;
+                            }).each(function(entry) {
+                                selfoss.db.storage.entries.delete(entry.id);
+                                if (selfoss.dbOffline.newestGCedEntry < entry.datetime) {
+                                    selfoss.dbOffline.newestGCedEntry = entry.datetime;
+                                }
+                            }
+                            ).then(function() {
+                                selfoss.db.storage.stamps.bulkPut([
+                                    {name: 'lastCleanup', datetime: new Date()},
+                                    {
+                                        name: 'newestGCedEntry',
+                                        datetime: selfoss.dbOffline.newestGCedEntry
+                                    }
+                                ]);
+                            });
+                    }
+                });
+            });
+    },
+
+
+    storeStats: function(stats) {
+        return selfoss.dbOffline._tr('rw', selfoss.db.storage.stats,
+            function() {
+                for (var stat in stats) {
+                    if (stats.hasOwnProperty(stat)) {
+                        selfoss.db.storage.stats.put({
+                            name: stat,
+                            value: stats[stat]
+                        });
+                    }
+                }
+            });
+    },
+
+
+    storeLastUpdate: function(lastUpdate) {
+        return selfoss.dbOffline._tr('rw', selfoss.db.storage.stamps,
+            function() {
+                if (lastUpdate) {
+                    selfoss.db.storage.stamps.put({
+                        name: 'lastItemsUpdate',
+                        datetime: lastUpdate
+                    });
+                }
+            });
+    },
+
+
+    reloadList: function() {
+        return selfoss.dbOffline._tr('r', selfoss.db.storage.entries,
+            function() {
+                var content = $('#content');
+                var newContent = content.clone().empty();
+
+                var howMany = 0;
+                var hasMore = false;
+
+                var ascOrder = selfoss.db.ascOrder();
+                var entries = selfoss.db.storage.entries.orderBy('[datetime+id]');
+                if (!ascOrder) {
+                    entries = entries.reverse();
+                }
+
+                var seek = false;
+                var fromDatetime = selfoss.filter.fromDatetime;
+                var fromId = selfoss.filter.fromId;
+                if (fromDatetime && fromId) {
+                    seek = true;
+                    fromDatetime = new Date(fromDatetime);
+                }
+                var isMore = false;
+                var alwaysInDb = selfoss.filter.type == 'starred'
```

Comment:
We should use strict comparison when it works, it will make getting rid of the weak one easier.